fredrick-kamau1/banking-app
# Prepare DBTrans class for dependency injection

TODO: Test to avoid exceptions

* Implement a new interface IDBTrans
* Ensure all callers are using the interface instead of the implementation class.
* Instantiate DBTrans using try-with-resources.
* Allow DBTrans to control the lifetime of its own SQL Connection, and to share the same connection (implicitly) across all methods
* Remove connection parameter from all methods of IDBTrans.
* Allow DBTrans.createDB to control the lifetime of its own SQL Statement, and remove it from the parameters.
* Build fixes for gradle and IntelliJ IDEA on my system
